src/npc.cpp

```diff
@@ -1624,6 +1628,89 @@ void npc::say( const std::string line, ... ) const
     }
 }
 
+void npc::hear_sound( const tripoint &source, float volume, const std::string &verbal_shout_id )
+{
+    if( is_deaf() ) {
+        return;
+    }
+
+    // Estimating loudness falloff linearly with distance as done elsewhere.
+    const int dist = rl_dist( source, pos() );
+    volume -= dist;
+    // TODO: Could factor in weather, terrain, temperature, ambient sound, etc
+    // TODO: More severe attenuation shenanigans for diff underground z axis?
+    volume *= hearing_ability();
+
+    // 2 is the quietest perceptible sound
+    if( volume < 2 ) {
+        return;
+    }
+
+    // Sound localization error: proportional to distance, and inversely proportional
+    // to loudness. Clamp to distance so target is never in opposite direction
+    const bool sees_source = sees(source);
+    const int localization_error = sees_source ? 0 : std::min( dist / volume, (float)dist );
+    const int target_x = source.x + rng(-localization_error, localization_error);
+    const int target_y = source.y + rng(-localization_error, localization_error);
+    const int target_z = source.z;
+
+    if( verbal_shout_id == "SHOUT_CALL" ) {
```

Comment:
It's also used as a sound event tag for the soundpack subsystem. I added a new commit that replaces the string with an enum anyhow, to see how it looks. After looking at the new commit, if you prefer the string over enum, it's easy enough to remove the commit.